File: admin_portal/src/app/enum/DiscountType.ts
export class DiscountType{

  readonly key: string;
  readonly value: string;

  constructor(key: string, value: string) {
      this.key = key;
      this.value = value;
  }

  public static Percentage = new DiscountType("Percentage", "Percentage");
  public static Consolidate = new DiscountType("Consolidate", "Consolidate");
 
  static getAllEnumList(): Array<DiscountType> {
    return [
      DiscountType.Percentage,
      DiscountType.Consolidate,
    ];
  }

}

File: ecommerce_api/src/main/java/com/example/ecommerce_api/enums/DiscountType.java
package com.example.ecommerce_api.enums;

import com.example.ecommerce_api.exception.InvalidEnumValueException;

public enum DiscountType {

  Percentage("Percentage","Percentage"),
  Consolidate("Consolidate","Consolidate");

  private final String key;
  private final String value;

  DiscountType(String key,String value) {
    this.key = key;
    this.value = value;
  }

  public String getKey() {
    return key;
  }
  public String getValue() {
    return value;
  }

  public static DiscountType getEnumByValue(String value) {
    for (DiscountType item : DiscountType.values()) {
      if (item.getValue().equals(value)) {
        return item;
      }
    }
    throw new InvalidEnumValueException();
  }
}
